crates/ruff_formatter/src/format_element/tag.rs

```diff
@@ -375,6 +375,32 @@ pub trait LabelDefinition {
     fn name(&self) -> &'static str;
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+/// Stores data about a line suffix tag.
+pub struct LineSuffix {
+    pub(crate) reserved_width: u32,
+}
+
+impl LineSuffix {
+    pub fn new() -> Self {
+        LineSuffix {
+            reserved_width: Default::default(),
+        }
+    }
+
+    #[must_use]
+    pub fn with_width(mut self, width: u32) -> Self {
+        self.reserved_width = width;
+        self
+    }
+}
+
+impl Default for LineSuffix {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
```

Comment:
Nit: We don't have named structs for most tag variants because this is mostly an internal API, but being able to extract the fields can be useful. But I do see how `StartLineSuffix(u32)` isn't very meaningful.

We could introduce a `TextWidth(u32)` newtype wrapper that also offers `from_text`, `from_char` methods that then allow code-reuse between the Printer and client code.

Reply:
Yea this did feel out of place when I added it. Thanks for the feedback! I really like the `TextWidth` idea.

Reply:
Would this be something we'd want to add to `ruff_text_size` or similar? Not sure if I'm overthinking it, but I'm trying to understand the `TextSize` struct and how `TextWidth` might relate.

https://github.com/astral-sh/ruff/blob/15b7525464e3f7bd1d28d69a5feb0d50913906e2/crates/ruff_text_size/src/size.rs#L12-L27

This caught my eye
https://github.com/astral-sh/ruff/blob/15b7525464e3f7bd1d28d69a5feb0d50913906e2/crates/ruff_text_size/src/lib.rs#L9-L15

Reply:
I would keep it in `ruff_formatter` for now because it will need access to `TabWidth` which is formatter specific too.

Reply:
Removed the named struct for `StartLineSuffix(u32)` for now. I do want to implement `TextWidth`, but it's possible I'll try to shoot to land a minimum PR instead. If that changes I'll scoop it up here.